evm/v0.5/contracts/dev/PrepaidAggregator.sol

```diff
@@ -0,0 +1,242 @@
+pragma solidity 0.5.0;
+
+import "../Median.sol";
+import "../vendor/Ownable.sol";
+import "../vendor/SafeMath.sol";
+import "./SafeMath128.sol";
+import "../interfaces/LinkTokenInterface.sol";
+
+/**
+ * @title The PrepaidAggregator handles aggregating data pushed in from off-chain.
+ */
+contract PrepaidAggregator is Ownable {
+  using SafeMath for uint256;
+  using SafeMath128 for uint128;
+
+  struct OracleStatus {
+    bool enabled;
+    uint128 withdrawable;
+    uint256 lastReportedRound;
+  }
+
+  struct Round {
+    uint64 maxAnswers;
+    uint64 minAnswers;
+    uint128 paymentAmount;
+    int256[] answers;
+  }
+
+  int256 public currentAnswer;
+  uint256 public currentRound;
+  uint128 public paymentAmount;
+  uint64 public oracleCount;
+  uint64 public maxAnswerCount;
+  uint64 public minAnswerCount;
+  uint256 public updatedHeight;
+  uint128 public availableFunds;
+  uint128 public allocatedFunds;
+
+  LinkTokenInterface private LINK;
+  mapping(address => OracleStatus) private oracles;
+  mapping(uint256 => Round) private rounds;
+
+  event NewRound(uint256 indexed number, address indexed startedBy);
+  event AnswerUpdated(int256 indexed current, uint256 indexed round);
+  event AvailableFundsUpdated(uint256 indexed amount);
+
+  constructor(address _link, uint128 _paymentAmount) public {
+    LINK = LinkTokenInterface(_link);
+    setPaymentAmount(_paymentAmount);
+  }
+
+  function updateAnswer(uint256 _round, int256 _answer)
+    public
+    ensureValidRoundId(_round)
+    validateOracleRound(_round)
+  {
+    startNewRound(_round);
+    recordAnswer(_answer, _round);
+    updateRoundAnswer(_round);
+    payOracle(_round);
+    deleteRound(_round);
+  }
+
+  function addOracle(address _oracle)
+    public
+    onlyOwner()
+    ensureNotEnabledAddress(_oracle)
+  {
+    require(oracleCount < 42, "cannot add more than 42 oracles");
```

Comment:
Good point. It's a magic number in a couple of ways. I think 49 was the limit within the default gas price that a Chainlink node provides. We should run some tests and get a better idea, better idea of what the limits actually are here, because I bet they're higher now.

Reply:
I'm going to mark this as resolved, since it sounds like there's work here beyond the scope of this story.